lambda/main.py

```diff
@@ -1,73 +1,158 @@
+
+    try:
+        response = dynamodb.describe_table(TableName=table_name)
+        table_status = response['Table']['TableStatus']
+    except Exception as e:
+        if hasattr(e, 'response') and 'Error' in e.response:
+            if e.response['Error']['Code'] != 'ResourceNotFoundException':
+                logger.error(f"Unknown exception raised on DescribeTable API: {e.response}")
+                return jsonify({
+                    'status': 'error',
+                    'errors': 'Unknown exception',
+                }), 501
+
+        return jsonify({
+                    'status': 'error',
+                    'errors': f"Table {table_name} does not exist, try running \"eden config --push\"",
+                }), 501
+
+    if table_status != 'ACTIVE':
+        return jsonify({
+                    'status': 'error',
+                    'errors': f"Table {table_name} status is \"{table_status}\", try again later",
+                }), 501
+
+    r = dynamodb.get_item(
+        TableName=table_name,
+        Key={
+            'env_name': {
+                'S': f"_profile_{g.profile}",
+            }
+        },
+    )
+
+    if 'Item' in r:
+        if 'profile' in r['Item']:
+            try:
+                g.config = json.loads(r['Item']['profile']['S'])
+            except json.JSONDecodeError:
+                logger.error(f"Profile {g.profile} is invalid: {r['Item']['profile']}")
+                return jsonify({
+                    'status': 'error',
+                    'errors': f"Profile {g.profile} is invalid",
+                }), 501
+        else:
+            return jsonify({
+                'status': 'error',
+                'errors': f"Profile {g.profile} exists, but is empty",
+            }), 501
+    else:
+        return jsonify({
+            'status': 'error',
+            'errors': f"Profile {g.profile} not found",
+        }), 404
+
+
 @app.route('/api/v1/create')
 def create_env():
-    if 'branch' not in request.args or 'image_uri' not in request.args:
+    if 'name' not in request.args or 'image_uri' not in request.args:
         return jsonify({
             'status': 'error',
             'reason': 'Insufficient query parameters',
-            'code':   400,
         }), 400
 
-    branch: str = request.args.get('branch')
+    name: str = request.args.get('name')
     image_uri: str = request.args.get('image_uri')
-    errors = eden_core.methods.create_env(branch, image_uri, None)
-
-    if not errors:
-        return jsonify({
-            'status': 'ok',
-            'code':   200
-        }), 200
-    else:
+    # noinspection PyBroadException
+    try:
+        r = aws_eden_core.methods.create_env(name, image_uri, g.config)
+    except Exception as e:
+        logger.error(e)
         return jsonify({
             'status': 'error',
-            'code':   501
+            'errors': f"Exception caught",
         }), 501
 
+    table = boto3.resource('dynamodb').Table(table_name)
```

Comment:
I'm not sure which one is preferable, but could you align the way of handling DynamoDB client?

I found this code on the upper side:

https://github.com/baikonur-oss/terraform-aws-lambda-eden-api/blob/c1a1865e10c982d15a09b844a4a5c00742b8ca32/lambda/main.py#L60-L67

Reply:
dynamodb is boto3 DynamoDB client, while `boto3.resource('dynamodb')` is boto3 DynamoDB resource. These are not the same, and it is difficult to name them. 

We can go with dynamodb_client for former, and dynamodb_resource for latter, how does that sound?

Reply:
Ah, sorry for ignorance. That would be easier to understand :)

Reply:
Will rename here and API parts then.

Reply:
Done for both